### PR TITLE
test: add API key tenant field tests for MaaS

### PR DESCRIPTION
**Pull request by EgorLu**
# Pull Request

## Summary

Add tests verifying the `tenant` field is present and defaults to empty string in API key responses (POST create, GET by ID, POST search).

## Related Issues

- JIRA: RHOAIENG-66549
- Dev PR: https://github.com/opendatahub-io/models-as-a-service/pull/951

### Tests added (`test_api_key_tenant_field.py`)

1. **test_create_api_key_response_includes_tenant_field** (smoke) -- POST /v1/api-keys response includes tenant field defaulting to empty string.
2. **test_get_api_key_response_includes_tenant_field** (tier1) -- GET /v1/api-keys/{id} response includes tenant field defaulting to empty string.
3. **test_list_api_keys_all_items_include_tenant_field** (tier1) -- POST /v1/api-keys/search items all include tenant field defaulting to empty string.

### Utils added

- **assert_tenant_field_empty** -- Reusable assertion helper verifying tenant field presence and empty-string value in any key response body.

## Please review and indicate how it has been tested

- [ ] Locally
- [ ] Jenkins

## Additional Requirements

- [ ] If this PR introduces a new test image, did you create a PR to mirror it in disconnected environment?
- [ ] If this PR introduces new marker(s)/adds a new component, was relevant ticket created to update relevant Jenkins job?

## Summary by CodeRabbit

* **Tests**
  * Added coverage to verify MaaS API key responses include a `tenant` field defaulting to an empty string for key creation, retrieval by ID, and listing/search results.
  * Added a dedicated test suite to validate the field is present and correct across all returned items.
* **Tests**
  * Introduced a shared assertion helper to consistently check the `tenant` field in API responses.

### CODE REVIEW

**Review by dbasunag — DISMISSED** [review]

File: tests/model_serving/maas_billing/maas_api_key/test_api_key_tenant_field.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+def assert_tenant_field_empty(body: dict, context: str) -> None:
+    """Assert that the response body contains a 'tenant' field defaulting to empty string.
+
+    Args:
+        body: Parsed JSON response body from a MaaS API key endpoint.
+        context: Human-readable label for assertion error messages (e.g. "POST /v1/api-keys").
+    """
+    assert "tenant" in body, f"Expected 'tenant' field in {context} response, got keys: {list(body.keys())}"
+    assert body["tenant"] == "", f"Expected tenant='' (empty string) in {context} response, got: {body['tenant']!r}"
+
```
Comment (SB159) [inline tests/model_serving/maas_billing/maas_api_key/test_api_key_tenant_field.py]: Move this function to tests/model_serving/maas_billing/maas_api_key/utils.py and import it. as per project guidelines, all reusable utility/assertion functions belong in utils.py, not in test files.

File: tests/model_serving/maas_billing/maas_api_key/test_api_key_tenant_field.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        self,
+        request_session_http: requests.Session,
+        base_url: str,
+        ocp_token_for_actor: str,
+    ) -> None:
+        """Verify POST /v1/api-keys response includes tenant field defaulting to empty string."""
+
+        key_name = f"e2e-tenant-create-{generate_random_name()}"
+
+        resp, body = create_api_key(
```
Comment (SB159) [inline tests/model_serving/maas_billing/maas_api_key/test_api_key_tenant_field.py]: create_api_key() is called directly in the test body with no teardown — the key is never revoked. Use the existing active_api_key_with_plaintext fixture from conftest.py instead; it handles creation, cleanup, and yields the full response body including the tenant field. The assert_api_key_created_ok call is also redundant since create_api_key() already raises on non-200/201.

**Review by SB159 — CHANGES_REQUESTED** [review]

**Review by SB159 — APPROVED** [review]
/lgtm

**Review by dbasunag — APPROVED** [review]
